[iOS] Apple mobile NativeAOT test suits stopped building: `task could not be loaded from the assembly`

## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=1016183
Build error leg or test failing: Build
Affected CI: all Apple mobile NativeAOT jobs
The range of commits: https://github.com/dotnet/runtime/compare/159f40e8...a58e9239

Log snippet:
```
2025-04-14T22:10:47.7365510Z ##[error]artifacts/bin/coreclr/tvos.arm64.Release/build/Microsoft.NETCore.Native.Publish.targets(74,5): error MSB4062: The "ComputeManagedAssembliesToCompileToNative" task could not be loaded from the assembly /Users/runner/work/1/s/artifacts/bin/coreclr/tvos.arm64.Release/ilc-published/netstandard/ILCompiler.Build.Tasks.dll. Could not load file or assembly '/Users/runner/work/1/s/artifacts/bin/coreclr/tvos.arm64.Release/ilc-published/netstandard/ILCompiler.Build.Tasks.dll'. The system cannot find the file specified. [/Users/runner/work/1/s/src/tests/nativeaot/SmokeTests/UnitTests/UnitTests.csproj]
2025-04-14T22:10:47.7397870Z /Users/runner/work/1/s/artifacts/bin/coreclr/tvos.arm64.Release/build/Microsoft.NETCore.Native.Publish.targets(74,5): error MSB4062: Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/Users/runner/work/1/s/src/tests/nativeaot/SmokeTests/UnitTests/UnitTests.csproj] [/Users/runner/work/1/s/src/tests/build.proj]
```

<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorPattern": "artifacts\\/bin\\/coreclr\\/(?:tvos|ios|tvossimulator|iossimulator)\\.arm64\\.Release\\/build\\/Microsoft\\.NETCore\\.Native\\.Publish\\.targets.+The.*task could not be loaded from the assembly ",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=1016183
**Error message validated:** `[artifacts\/bin\/coreclr\/(?:tvos|ios|tvossimulator|iossimulator)\.arm64\.Release\/build\/Microsoft\.NETCore\.Native\.Publish\.targets.+The.*task could not be loaded from the assembly `]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 4/16/2025 11:36:41 AM UTC
<!-- Known issue validation end -->
<!--Known issue error report start -->

### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[1019283](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1019283)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/1019283/logs/494)||
|[1018649](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1018649)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/1018649/logs/491)||
|[1018037](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1018037)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/1018037/logs/432)||
|[1016183](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1016183)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/1016183/logs/435)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|2|4|4|
<!--Known issue error report end -->

--- a/eng/Subsets.props
+++ b/eng/Subsets.props
@@ -440,12 +440,12 @@
                              $(CoreClrProjectRoot)tools\PdbChecker\PdbChecker.csproj;
                              $(CoreClrProjectRoot)tools\AssemblyChecker\AssemblyChecker.csproj;
                              $(ToolsProjectRoot)StressLogAnalyzer\src\StressLogAnalyzer.csproj" Category="clr" Condition="'$(DotNetBuildSourceOnly)' != 'true'"/>
-    <!-- skip the architectures that don't have LKG runtime packs -->
-    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2_publish.csproj" Condition="'$(AotToolsSupported)' == 'true'" Category="clr" />
-    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler.Build.Tasks\ILCompiler.Build.Tasks.csproj" Category="clr" Condition="'$(AotToolsSupported)' == 'true'" />
-    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler\ILCompiler_publish.csproj" Category="clr" Condition="'$(AotToolsSupported)' == 'true'" />
     <!-- We may use these targets with the in-build ILCompiler or with the shipping ILCompiler. Produce them whenever we may use them. -->
     <ProjectToBuild Include="$(CoreClrProjectRoot)nativeaot\BuildIntegration\BuildIntegration.proj" Category="clr" Condition="'$(AotToolsSupported)' == 'true' or '$(NativeAotSupported)' == 'true'" />
+    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler.Build.Tasks\ILCompiler.Build.Tasks.csproj" Category="clr" Condition="'$(AotToolsSupported)' == 'true' or '$(NativeAotSupported)' == 'true'" />
+    <!-- skip the architectures that don't have LKG runtime packs -->
+    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2_publish.csproj" Condition="'$(AotToolsSupported)' == 'true'" Category="clr" />
+    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler\ILCompiler_publish.csproj" Category="clr" Condition="'$(AotToolsSupported)' == 'true'" />
 
     <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler\ILCompiler.csproj" Condition="'$(AotToolsSupported)' == 'true'" Category="clr" />
     <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2.csproj" Condition="'$(AotToolsSupported)' == 'true'" Category="clr" />

